mkosi/__init__.py

```diff
@@ -2159,6 +2112,42 @@ def needs_build(args: MkosiArgs, config: MkosiConfig) -> bool:
     return args.verb in MKOSI_COMMANDS_NEED_BUILD and (args.force > 0 or not config.output_dir.joinpath(config.output_with_compression).exists())
 
 
+def load_dynamic_modules() -> None:
+    """
+    Because we overmount /usr when using a tools tree, we need to make sure we load all python modules we
+    might end up using before overmounting /usr. Any modules that might be dynamically loaded during
+    execution are forcibly loaded early using this function.
+    """
+
+    try_import("importlib.readers")
+    try_import("importlib.resources.readers")
```

Comment:
More of a nit: Any reason why the `readers` submodules are imported specifically? The modules are public, but they're not documented - at least I didn't find them in the docs right away - so they might be an implementation detail? Maybe there's something that can be imported that's a documented public API?

Reply:
It's the one we need, I'm happy to accept alternatives but I wouldn't know what else to import